cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -122,15 +161,23 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    * @throws NotFoundException if either the namespace or dataset instance is not found,
    * @throws IOException if there is a problem in making an HTTP request to check if the namespace exists.
    */
-  public DatasetMeta get(Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
+  public DatasetMeta get(final Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
```

Comment:
That is automatically done in ``HttpExceptionHandler``. It will be 403, though, if authorization fails